crates/core/src/rpc/full.rs

```diff
@@ -1843,47 +1849,8 @@ impl Full for SurfpoolFullRpc {
                 )));
             }
 
-            let remote_slots = if let (Some((remote_client, _)), Some(local_min)) =
-                (&remote_ctx, local_min_slot)
-            {
-                if start_slot < local_min {
-                    let remote_end = effective_end_slot.min(local_min.saturating_sub(1));
-                    if start_slot <= remote_end {
-                        remote_client
-                            .client
-                            .get_blocks(start_slot, Some(remote_end))
-                            .await
-                            .unwrap_or_else(|_| vec![])
-                    } else {
-                        vec![]
-                    }
-                } else {
-                    vec![]
-                }
-            } else if remote_ctx.is_some() && local_min_slot.is_none() {
-                remote_ctx
-                    .as_ref()
-                    .unwrap()
-                    .0
-                    .client
-                    .get_blocks(start_slot, Some(effective_end_slot))
-                    .await
-                    .unwrap_or_else(|_| vec![])
-            } else {
-                vec![]
-            };
-
-            // Combine results
-            let mut combined_slots = remote_slots;
-            combined_slots.extend(local_slots);
-            combined_slots.sort_unstable();
-            combined_slots.dedup();
-
-            if combined_slots.len() > MAX_SLOT_RANGE as usize {
-                combined_slots.truncate(MAX_SLOT_RANGE as usize);
-            }
-
-            Ok(combined_slots)
+            let slots: Vec<Slot> = (start_slot..=effective_end_slot).collect();
+            Ok(slots)
```

Comment:
I'm pretty sure all of these changes to get_blocks should be reverted except the changes we made to the commitment config